extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/DataSourceBuildTimeConfig.java

```diff
@@ -22,4 +22,11 @@
     @ConfigItem(defaultValue = "enabled")
     public TransactionIntegration transactions;
 
+    /**
+     * Enable datasource metrics collection. If unspecified, collecting metrics will be enabled by default if the
+     * smallrye-metrics extension is active.
+     */
+    @ConfigItem
+    public Optional<Boolean> enableMetrics;
```

Comment:
So we don't have a way to disable metrics at runtime?

It seems like a very common request.

Reply:
Agroal seems to support disabling metrics dynamically, but I made this a build time property after discussing with Emmanuel who thinks we don't need that: https://github.com/quarkusio/quarkus/pull/5272#issuecomment-555106563

Reply:
I don't know if we should actually be able to turn it off at runtime given everything the user has to do to actually turn it on :). I mean it seems like the user really wants it explicitly. 

Now I am sure there scenaria were someone will want to turn it off, but perhaps we can keep it as is for now and reexamine if we have such a request in the future?

Reply:
Yeah I guess, I don't see much usefulness in dynamically turning off something that you had to explicitly turn on in the first place. 
And especially since turning off metric collection internally will mean that the already collected exposed metrics will suddenly "freeze" at their current values, which can become a pitfall, unless we add a hook that will also de-register the externally exposed Metrics as soon as this gets turned off.

Reply:
So FTR, my use case was: metrics are not free, if you're in a constrained environment, you might want to disable them most of the time and enable them in case something is going wrong.

But yes, it can probably wait.